org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/jres/JREsPreferencePage.java

```diff
@@ -131,7 +135,24 @@ protected Control createContents(Composite ancestor) {
 		SWTFactory.createWrapLabel(ancestor, JREMessages.JREsPreferencePage_2, 1, 300);
 		SWTFactory.createVerticalSpacer(ancestor, 1);
 
-		detectAtStartupCheckbox = SWTFactory.createCheckButton(ancestor, JREMessages.detectJREsAtStartup, null, getPreferenceStore().getBoolean(LaunchingPlugin.PREF_DETECT_VMS_AT_STARTUP), 1);
+		Composite detectButtons = SWTFactory.createComposite(ancestor, 2, 1, 0);
+		detectJvmButton = SWTFactory.createPushButton(detectButtons, JREMessages.JREsPreferencePage_DetectJVMsNow, null);
+		detectJvmButton.addListener(SWT.Selection, new Listener() {
+			@Override
+			public void handleEvent(Event evt) {
+				DetectVMInstallationsJob job = new DetectVMInstallationsJob();
+				job.schedule();
+				try {
+					job.join();
```

Comment:
Isn't there a risk of freezing the UI here? If so, I would recommend that the button should just schedule the job (and disable itself), and that a job listener gets added to react upon completion.